Sources/SwiftDocC/Utility/Errors/FileSystemError.swift

```diff
@@ -0,0 +1,21 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+public enum FileSystemError: Error, DescribedError {
```

Comment:
Is there an error from Foundation that we can use instead of defining a new error for this?

Reply:
I'm not aware of such. Additionally, all errors in `Foundation` bridge to `NSError`/`CFError`, which is a reference type and are much heavier than a plain enum declared here. Neither of those bridged types are declared in [`swift-foundation`](https://github.com/apple/swift-foundation), thus when switching from `swift-corelibs-foundation` to `swift-foundation` in the future that's one more thing to account for, while using our own enum means we can avoid such migration.

Reply:
Wouldn't we be able to use one of these errors? https://github.com/apple/swift-foundation/blob/main/Sources/FoundationEssentials/ErrorCodes.swift

Reply:
Is there a specific one that you had in mind? I see some for file not existing, or lack of reading permissions, but the newly created `FileSystemError.noDataReadFromFile` is specifically for when the file exists, is readable, but `readToEnd()` returns `nil`. I don't see a corresponding error code for that in Foundation.

Reply:
AFAICT `fileReadUnknown` is the same as `NSFileReadUnknownError` which is described as 
> Could not read, for unknown reasons.

That seems to fit here.